django/db/models/query.py

```diff
@@ -843,6 +843,10 @@ def select_related(self, *fields):
 
         If select_related(None) is called, the list is cleared.
         """
+
+        if self._fields is not None:
+            raise TypeError("Cannot call select_related() after .values() or .values_list()")
```

Comment:
I think `ValueError` is the more appropriate error here. As you've pointed out in IRC though, other examples, like `delete()` and `only()`, use this exact logic and raise type errors. It's better to remain consistent.

I don't think we should change the others to ValueError either, because that'd break user code for no real gain.